app/controllers/application_controller.rb

```diff
@@ -4,4 +4,14 @@ class ApplicationController < ActionController::Base
   rescue_from CanCan::AccessDenied do |exception|
     redirect_to root_url, alert: exception.message
   end
+
+  private
+
+  def after_sign_in_path_for(_)
+    authenticated_root_path
+  end
+
+  def after_sign_up_path_for(_)
+    authenticated_root_path
+  end
```

Comment:
Yeah if you don't use this and change root path to devise sign in it will enter a loop and you won't go to any page, so you need to specify where you want to be redirected after sign_up/in.